openaerostruct/functionals/moment_coefficient.py

```diff
@@ -216,28 +264,62 @@ def compute_partials(self, inputs, partials):
-            partials["CM", name + "_S_ref"] = -np.outer(M_j, dMAC_dS * term)
 
             # For first surface, we need to save the deriv results
             if j == 0:
+                # Compute the derivative of CM wrt to the chord distribution by taking the negative outer product of the
+                # moment vector(M_j) time the term with the derivative of MAC wrt to the chord distribution. We only do
+                # this for the main wing since CM only depends on the MAC of the main wing and the chord distribution of
+                # the main wing is the only chord distribution of all the surfaces that can impact the MAC of the main wing.
+                partials["CM", name + "_chords"] = -np.outer(M_j * term, dMAC_dc)
+
+                # Compute the derivative of CM wrt to the width distribution by taking the negative outer product of the
+                # moment vector(M_j) time the term with the derivative of MAC wrt to the width distribution. We only do
+                # this for the main wing since CM only depends on the MAC of the main wing and the panel width distribution of
+                # the main wing is the only panel width distribution of all the surfaces that can impact the MAC of the main wing.
+                partials["CM", name + "_widths"] = -np.outer(M_j * term, dMAC_dw)
+
+                # Compute the derivative of CM wrt to the surface S_ref by taking the negative outer product of the
+                # moment vector(M_j) time the term with the derivative of MAC wrt to the surface S_ref. The CM depends on
+                # the total references area of all surfaces including the main wing and the MAC of them main wing itself
+                # As result, this derivative has two parts only for the main wing.
+                # partials["CM", name + "_S_ref"] = -np.outer(M_j, dMAC_dS * term)
+                partials["CM", name + "_S_ref"] = np.outer(M_j * fact, (-1/S_ref_total + 1/S_ref))
+
+                # Cache the main wing's MAC derivatives
                 base_name = name
                 base_dMAC_dc = dMAC_dc
                 base_dMAC_dw = dMAC_dw
-                base_dMAC_dS = dMAC_dS
-
+                # base_dMAC_dS = dMAC_dS
             else:
                 # Apply this surface's portion of the moment to the MAC_wing term.
+                # We need to do this because CM is normalized by the MAC of the main wing
                 term = fact / (MAC_wing * MAC)
                 partials["CM", base_name + "_chords"] -= np.outer(M_j * term, base_dMAC_dc)
                 partials["CM", base_name + "_widths"] -= np.outer(M_j * term, base_dMAC_dw)
-                partials["CM", base_name + "_S_ref"] -= np.outer(M_j, base_dMAC_dS * term)
+                # partials["CM", base_name + "_S_ref"] -= np.outer(M_j, base_dMAC_dS * term)
+                partials["CM", base_name + "_S_ref"] = np.outer(M_j * fact, (-1/S_ref_total + 1/S_ref_wing))
```

Comment:
This change and also why `term` has `MAC` and `MAC_wing` in the denominator in this `else` branch isn't totally clear to me. Could you explain the change in more detail?

Reply:
I actually don't think that is correct and I just fixed it